catalog/app/containers/Bucket/Overview/Downloads.tsx

```diff
@@ -0,0 +1,593 @@
+import cx from 'classnames'
+import * as dateFns from 'date-fns'
+import * as Eff from 'effect'
+import * as React from 'react'
+import * as M from '@material-ui/core'
+import { fade } from '@material-ui/core/styles'
+import useComponentSize from '@rehooks/component-size'
+
+import StackedAreaChart from 'components/StackedAreaChart'
+import cfg from 'constants/config'
+import * as GQL from 'utils/GraphQL'
+import log from 'utils/Logging'
+import * as SVG from 'utils/SVG'
+import { readableQuantity } from 'utils/string'
+
+import { ColorPool } from './ColorPool'
+
+import BUCKET_ACCESS_COUNTS_QUERY from './gql/BucketAccessCounts.generated'
+
+type GQLBucketAccessCounts = NonNullable<
+  GQL.DataForDoc<typeof BUCKET_ACCESS_COUNTS_QUERY>['bucketAccessCounts']
+>
+type GQLAccessCountsGroup = GQLBucketAccessCounts['byExt'][0]
+type GQLAccessCounts = GQLBucketAccessCounts['combined']
+type GQLAccessCountForDate = GQLAccessCounts['counts'][0]
+
+interface ProcessedAccessCountForDate {
+  date: Date
+  value: number
+  sum: number
+}
+
+interface ProcessedAccessCounts {
+  total: number
+  counts: readonly ProcessedAccessCountForDate[]
+}
+
+interface ProcessedAccessCountsGroup {
+  ext: string
+  counts: ProcessedAccessCounts
+}
+
+interface ProcessedBucketAccessCounts {
+  byExt: readonly ProcessedAccessCountsGroup[]
+  byExtCollapsed: readonly ProcessedAccessCountsGroup[]
+  combined: ProcessedAccessCounts
+}
+
+const processAccessCountForDateArr = (
+  counts: readonly GQLAccessCountForDate[],
+): readonly ProcessedAccessCountForDate[] =>
+  // compute running sum
+  Eff.Array.mapAccum(counts, 0, (acc, { value, date }) => [
+    acc + value,
+    {
+      value,
+      date,
+      sum: acc + value,
+    },
+  ])[1]
+
+const processAccessCounts = (counts: GQLAccessCounts): ProcessedAccessCounts => ({
+  total: counts.total,
+  counts: processAccessCountForDateArr(counts.counts),
+})
+
+const processAccessCountsGroup = (
+  group: GQLAccessCountsGroup,
+): ProcessedAccessCountsGroup => ({
+  ext: group.ext && `.${group.ext}`,
+  counts: processAccessCounts(group.counts),
+})
+
+const processBucketAccessCounts = (
```

Comment:
yeah, but i don't see much value in this

Reply:
It's difficult to follow what is the structure of the output data. Test can show and guarantee it

Reply:
On the line 43 there is an interface defining the first level of the object. Then I need to go to the next level, then to the next. So, 3 levels to keep in mind.
`byExt` and `byExtCollapsed` re-use the same shape, so I need to check what is different in them, because I don't understand what collapsed means here. So one more thing to keep in.
Types are only about shape, but to have an idea what the data is I need to read the code. The code consist of multiple nested functions. I need to keep them in mind to get the idea of the result. I don't know what `Eff.Array.mapAccum`, so, I need to read the documentation and keep that in mind as well.

And somewhere in the process, I run out of focus, and start from the beginning. So, just looking at the results with sample data will help me.

Reply:
(i've only now pushed the actual commit with the test)

Reply:
https://github.com/quiltdata/quilt/pull/4223/files

I don't intend to merge it, just to show

Reply:
well, it kinda defeats the initial purpose, bc you have to consult several files to see the data

Reply:
Initial purpose?

For me, this particular data is more digestible in separate files or if it was defined somewhere independently, like, 
```
const input = {
...
}
const ouput = {
...
}
expect(input).toMatch(output)
```

When using files, also `diff` works.

---

But again, I'm ok with what is done and don't insist on some change. I keep conversation, because you do :)